### PR TITLE
[WIP] Wrap Magento's getUrl() with custom function

### PR DESCRIPTION
**Pull request by spgitmonish**
The custom function checks if the store/request is secure and then adds the "_secure" key with value set to true as an additional parameter.

### CODE REVIEW

File: app/code/community/Bolt/Boltpay/Helper/Api.php [hunk 1]
```diff
@@ -1166,4 +1166,21 @@ public function getIncrementIdFromTransaction( $transaction ) {
...
\ No newline at end of file
+
+    /**
+     * Generate (if) secure url by route and parameters
+     *
+     * @param   string $route
+     * @param   array $params
+     * @return  string
+     */
+    public function getSecureUrl($route = '', $params = array()){
```
Comment (daisy1754) [inline app/code/community/Bolt/Boltpay/Helper/Api.php]: I suggest you name this simply `getUrl` or `getMagentoUrl`. It's confusing if it is named secure but can return unsecure url.

Comment (daisy1754) [reply app/code/community/Bolt/Boltpay/Helper/Api.php]: Also suggest we don't add `$params` arg if we're not using today. YAGNI.

File: app/code/community/Bolt/Boltpay/Helper/Api.php [hunk 2]
```diff
@@ -1166,4 +1166,21 @@ public function getIncrementIdFromTransaction( $transaction ) {
...
+        if ((Mage::app()->getStore()->isFrontUrlSecure()) &&
+            (Mage::app()->getRequest()->isSecure())) {
+            // Connection is secure
```
Comment (daisy1754) [inline app/code/community/Bolt/Boltpay/Helper/Api.php]: redundant comment; suggest removing

File: app/design/adminhtml/default/default/template/boltpay/system/config/button.phtml [hunk 3]
```diff
@@ -32,7 +32,7 @@
 
         var xmlhttp = new XMLHttpRequest();
 
-        xmlhttp.open("POST", "<?=$this->getUrl('boltpay/configuration/check');?>", true);
+        xmlhttp.open("POST", "<?php echo Mage::helper('boltpay/api')->getSecureUrl('boltpay/configuration/check');?>", true);
```
Comment (daisy1754) [inline app/design/adminhtml/default/default/template/boltpay/system/config/button.phtml]: I'm actually not very familar with php. what is the difference between `<?=` and `<? echo`?

Comment (spgitmonish) [reply app/design/adminhtml/default/default/template/boltpay/system/config/button.phtml]: From what I read '<?=' implies '<?php echo>'

File: app/code/community/Bolt/Boltpay/Helper/Api.php (hunk 1 above)
Comment (spgitmonish) [reply app/code/community/Bolt/Boltpay/Helper/Api.php]: Will fix the naming.

File: app/design/adminhtml/default/default/template/boltpay/system/config/button.phtml (hunk 3 above)
Comment (daisy1754) [reply app/design/adminhtml/default/default/template/boltpay/system/config/button.phtml]: that's what I thought. If that's the case I'd say let's stick to  '<?=' so this diff won't contain any change other than replacing method (also '<?=' seems to be a majority across this plugin code base)

**Comment by daisy1754** [comment]
@spgitmonish Woops please don't merge unapproved PR. I just updated repository config so going forward merge button is disabled if PR is not approved..

**Pull request merged**